packages/inferno-animation/src/animationCoordinator.ts

```diff
@@ -0,0 +1,75 @@
+import { forceReflow } from './utils';
+
+export enum AnimationPhase {
+  INITIALIZE,
+  MEASURE,
+  SET_START_STATE,
+  ACTIVATE_TRANSITIONS,
+  REGISTER_LISTENERS,
+  ACTIVATE_ANIMATION
+}
+
+let _animationQueue: Function[] = [];
+let _animationActivationQueue: Function[] = [];
+let _nextAnimationFrame;
+let _nextActivateAnimationFrame;
+
+function _runActivateAnimationPhase() {
+  _nextActivateAnimationFrame = undefined;
+  // Get animations to execute
+  const animationQueue = _animationActivationQueue;
+  // Clear global queue
+  _animationActivationQueue = [];
+  
+  for (let i = 0; i < animationQueue.length; i++) {
+    animationQueue[i](AnimationPhase.ACTIVATE_ANIMATION);
+  }
+}
+
+function _runAnimationPhases() {
+  _nextAnimationFrame = undefined;
+  // Get animations to execute
+  const animationQueue = _animationQueue;
+  // Clear global queue
+  _animationQueue = [];
+  
+  // So what this does is run the animation phases in order. Most of the phases are invoked
+  // by a simple call to all the registered callbacks. However:
+  // - ACTIVATE_TRANSITIONS require a reflow in order to not
+  // interfere with the previous setting of the animation start class
+  // - ACTIVATE_ANIMATION needs to be called async so the transitions actually fire,
+  // we choose to use an animation frame.
+  for (let i = 0; i < 6; i++) {
+    const phase = i as AnimationPhase;
+    if (phase !== AnimationPhase.ACTIVATE_ANIMATION) {
```

Comment:
I would invert this if so the code reads better:
```js
if (phase === AnimationPhase.ACTIVATE_ANIMATION) {
// Final phase - Activate animations
} else {
 ...
}
```

Reply:
@Havunen is there some better way to iterate over enums in TS? I was searching high and low, but it appears to be an open issue. The `i < 6` feels really bad.

Reply:
yeah I don't think there is any... well at least it's fast :P We can, of course, write tests to make sure it functions correctly

Reply:
FIXED in 05e88c657 I am using a switch here now because it was even easier to read.